scylla-cql/src/frame/frame_errors.rs

```diff
@@ -34,16 +34,16 @@ pub enum FrameError {
     VersionNotSupported(u8),
     #[error("Connection was closed before body was read: missing {0} out of {1}")]
     ConnectionClosed(usize, usize),
-    #[error("Frame decompression failed.")]
-    FrameDecompression,
-    #[error("Frame compression failed.")]
-    FrameCompression,
     #[error(transparent)]
     StdIoError(#[from] std::io::Error),
     #[error("Unrecognized opcode{0}")]
     TryFromPrimitiveError(#[from] TryFromPrimitiveError<u8>),
+    #[error("Snap compression error: {0}")]
+    SnapCompressError(snap::Error),
     #[error("Error compressing lz4 data {0}")]
     Lz4CompressError(#[from] lz4_flex::block::CompressError),
+    #[error("Snap decompression error: {0}")]
+    SnapDecompressError(snap::Error),
     #[error("Error decompressing lz4 data {0}")]
     Lz4DecompressError(#[from] lz4_flex::block::DecompressError),
```

Comment:
What would you suggest then? We could have a common `(De)CompressError` variant for both snap and lz4 - it would not provide any additional context, or maybe it could include a stringified error returned from the corresponding library (and maybe some enum of (de)compression kind - either lz4 or snap).

Reply:
Why not Arc<dyn Error>?

Reply:
My phone skipped a part of message - I meant `Arc<dyn Error>`.

Reply:
Yes, I got that. I somehow forgot about this possibility and was trying to come up with some other solution